qiskit/circuit/library/basis_change/qft.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Quantum Fourier Transform Circuit."""
+
+import numpy as np
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+
+
+class QFT(QuantumCircuit):
+    r"""Quantum Fourier Transform Circuit.
+
+    The Quantum Fourier Transform (QFT) on :math:`n` qubits is the operation
+
+    .. math::
+
+        |j\rangle \mapsto \frac{1}{2^{n/2}} \sum_{k=0}^{2^n - 1} e^{2\pi ijk / 2^n} |k\rangle
+
+    The circuit that implements this transformation can be implemented using Hadamard gates
+    on each qubit, a series of controlled-U1 (or Z, depending on the phase) gates and a
+    layer of Swap gates. The layer of Swap gates can in principle be dropped if the QFT appears
+    at the end of the circuit, since then the re-ordering can be done classically. They
+    can be turned off using the ``do_swaps`` attribute.
```

Comment:
in this particular implementation the swaps appear in the beginning of the circuit, which does not match this description. i actually think swaps at the end are better, is it possible to change the `_build` method to do that?